### ISSUE TITLE
uno-check update fails

### ISSUE THREAD

**Issue by butti71** [unoplatform/uno#18927]
## Current Behavior
By executing: 

dotnet tool update -g uno.check

this error is coming up:

Tool 'uno.check' failed to update due to the following:
The settings file in the tool's NuGet package is invalid: Settings file 'DotnetToolSettings.xml' was not found in the package.
Tool 'uno.check' failed to install. Contact the tool author for assistance.

## Expected Behavior
The newest version of uno-check should be installed.

## How to reproduce it (as minimally and precisely as possible)
Calling "dotnet tool update -g uno.check"

## Workaround
I tried to deinstall uno-check, but still the same behavior.

### Works on UWP/WinUI

None

## Environment information

* Platform: linux 6.8.0-49-generic (x64)
* VSCode version: 1.95.3
* dotnet version: 8.0.300
* C# Extension: 'ms-dotnettools.csharp' version 2.55.29
* C# Dev Kit Extension: 'ms-dotnettools.csdevkit' version 1.13.9
* Uno Extension: 'unoplatform.vscode' version 0.18.5

## Extension Status
Extension: Dev Kit Mode
Debugger: Ready
Hot-Reload: Inactive
XAML LSP: Ready

**Comment by jeromelaban** [comment]
Thanks for the report.

Can you run the following command `dotnet --info` in the same folder you tried to run `dotnet tool update`, then paste the results here?

**Comment by jeromelaban** [comment]
This is quite interesting. It seems to be related to https://github.com/dotnet/sdk/issues/38172, which only happens when the supporting SDK is not installed, but uno-check supports .NET 5, so it should work as-is, particularly as you have .NET 8 installed.

Do you recall how you installed .NET ?

Otherwise, you can install .NET 9 manually, though Ubuntu 24.04 does not have .NET 9 from the official feeds yet, so you'd have to [install it manually](https://learn.microsoft.com/en-us/dotnet/core/install/linux-scripted-manual#manual-install).

**Comment by ajpinedam** [comment]
@butti71  try creating a `global.json` file at the root directory of your project, targeting your project to use one of the  .net 8.0.* versions you have installed.

From the command line, you can do it with the following:

```bash
dotnet new global.json --sdk-version 8.0.404
```

This should create a file with the following structure

```json
{
  "sdk": {
    "version": "8.0.404"
  }
}
```

Where `8.0.404` is one of the dotnet **8.0.***  versions you have installed in that system. 

To discover the specific versions you have available from the CLI, run:

```bash
dotnet --list-sdks
```

This global.json file tells the compiler to use the SDK specified in the file. 

This should allow you to have .net9.0 installed while still compiling .net8.0 projects.

**Comment by jeromelaban** [comment]
> Do you have another idea where to look at to fix that uno-check problem?

Just to be sure, I create a new VM with ubuntu 24.04, installed dotnet the same way you did (sudo apt-get install -y dotnet-sdk-8.0), and `dotnet tool install -g uno.check`. It installed properly.

Is there anything else specific from your installation that is out of the ordinary?

**Comment by butti71** [comment]
**### For all who are reading this: This part of the conversation is only indirect related to the uno-check error above, it is on a different machine where uno-check is up and running!**

> @butti71 try creating a `global.json` file at the root directory of your project, targeting your project to use one of the .net 8.0.* versions you have installed.
> 
> From the command line, you can do it with the following:
> 
> ```shell
> dotnet new global.json --sdk-version 8.0.404
> ```
> 
> This should create a file with the following structure
> 
> ```json
> {
>   "sdk": {
>     "version": "8.0.404"
>   }
> }
> ```
> 
> Where `8.0.404` is one of the dotnet **8.0.*** versions you have installed in that system.
> 
> To discover the specific versions you have available from the CLI, run:
> 
> ```shell
> dotnet --list-sdks
> ```
> 
> This global.json file tells the compiler to use the SDK specified in the file.
> 
> This should allow you to have .net9.0 installed while still compiling .net8.0 projects.

Thanks for the answer :-)
I created the global.json including the .NET sdk version (8.0.110) which is installed on my machine.
I run uno-check again, but it seems not to update the needed Uno.Sdk:

` MSB4276: The default SDK resolver failed to resolve SDK "Uno.Sdk" because directory "/usr/lib/dotnet/sdk/8.0.110/Sdks/Uno.Sdk/Sdk" did not exist.`

I tried adding the latest uno.sdk version in global.json:

```
"msbuild-sdks": {
    "Uno.Sdk": "5.5.49",
  }
```
To update the workloads, I found an issue which was answered also by Jérôme [https://github.com/unoplatform/uno/discussions/15044](Support for .NET 8.0.200) which helped me to reload all the required packages and also uno.sdk :

```
dotnet workload clean --all
dotnet workload restore
```

In my case I needed also to install a special workload:

`sudo dotnet workload install wasi-experimental`

after all, my project loads and all dependencies seem to be updated! :-)

When I try to build the solution I get now all these ghost errors, which I'm now investigating in:

![image](https://github.com/user-attachments/assets/2382893d-0868-4019-9d57-6cdb5378a004)

Some warnings in the output window might help me here:

warning NU1604: Project dependency Uno.Microsoft.Xaml.Behaviors.WinUI.Managed does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. 

Investigating further, but help is appreciated :-)

If you want me to open up a new ticket, please let me know, because this is only indirect related to the uno-check problem.

For now it gave me the confidence to install .NET 9.0 in parallel, so the uno-check problem can be solved. I will write an update to that one asap.

**Comment by butti71** [comment]
> > Do you have another idea where to look at to fix that uno-check problem?
> 
> Just to be sure, I create a new VM with ubuntu 24.04, installed dotnet the same way you did (`sudo apt-get install -y dotnet-sdk-8.0`), and `dotnet tool install -g uno.check`. It installed properly.
> 
> Is there anything else specific from your installation that is out of the ordinary?

I installed NET 9.0 according to the [manual install description](https://learn.microsoft.com/en-us/dotnet/core/install/linux-scripted-manual#manual-install) you provided above. This seems to do the trick on my machine. Also the old NET 8.0 project is still able to compile and run.

Thanks for the hint and also the extra effort to try it out on a VM!

Maybe for others or if you see a pattern, here are my "specialties" on this machine which might have interfere with the above error (what I cannot imagine)I:
- Added CommunityToolkit support with `dotnet add package CommunityToolkit.WinUI.Collections --version 8.1.240916`
- PHP development with laravel framework in vscode including xdebugger
- Build and run docker image to provide LAMP stack with exposed host folders for mySQL and PHP

Thanks again Jérôme!

I wish you all the best and keep up the good work on UNO!

Björg

**Comment by jeromelaban** [comment]
Thanks for the update and details! Could you tell if installing with `dotnet tool` uno.check is still failing with the same error, or if using net9 fixed it?

**Comment by butti71** [comment]
> Thanks for the update and details! Could you tell if installing with `dotnet tool` uno.check is still failing with the same error, or if using net9 fixed it?

It is fixed with net9 now. uno-check is now updated to the latest version v1.27.2.0

**Comment by jeromelaban** [comment]
Thanks for the update!
